### PR TITLE
Rapid check test

### PR DESCRIPTION
**Pull request by tjingrant**
Supporting Extensive Numerical Test via RapidCheck
-----------------------------------------------------

Motivation behind this patch is to introduce extensive numerical value based unit tests; this is very important as we start to perform optimizations. Corner cases can arise as we specialize for a specific architecture parameter (like vector width). This patch generates extensive amount of value-based unit tests based on a simple, naive (and extremely slow) implementation of convolution, used to verify the correctness of our convolution lowering and optimization.

Patch summary:
- Introduce utility functions to manipulate Dynamic MemRefs.
- Refactor ExecutionSession into a C++-only ExecutionSession + a PyExecutionSession so that C++ programs don't have to deal with python-specific dependencies.
- Introduce convolution (without strides/dilation) numerical unit test using a simple reference implementation.

### CODE REVIEW

File: test/numerical/TestConv.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+#include "mlir/IR/Module.h"
+#include "llvm/Support/FileSystem.h"
+
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "src/MainUtils.hpp"
+#include "src/Runtime/ExecusionSession.hpp"
+
+using namespace std;
+
+int main() {
```
Comment (AlexandreEichenberger) [inline test/numerical/TestConv.cpp:18]: Could you add a comment on how this is invoked? I expected a loop going over various N, C, H,.... values; here I see only one. I would add a few lines on how main is invoked (repetitively, with varying values?) and if so, where is the driver of this function.

thanks

Comment (tjingrant) [reply test/numerical/TestConv.cpp:130]: Added a documentation page on testing to explain how numerical tests are structured in general. I also added examples of exhaustive testing for a small range of values.
